### PR TITLE
Add support for HAPI 17 in Middleware

### PR DESCRIPTION
**Pull request by mkochendorfer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,10 +161,35 @@ app.listen(3001);
 
 ### Hapi
 
+#### Version 17
```
Comment (IvanGoncharov) [inline README.md]: Maybe worth to add `and above`?

Comment (mkochendorfer) [reply README.md]: Done

**Comment by mkochendorfer** [comment]
Addresses https://github.com/APIs-guru/graphql-voyager/issues/65

File: example/middleware/hapi.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const Hapi = require('hapi');
 const { graphqlHapi } = require('graphql-server-hapi');
-const { hapi: voyagerMiddleware } = require('graphql-voyager/middleware');
+const { hapiLegacy: voyagerMiddleware } = require('graphql-voyager/middleware');
```
Comment (IvanGoncharov) [inline example/middleware/hapi.js]: Rename this to `hapiLegacy.js` and create a new `hapi.js`.

File: README.md (hunk 1 above)
Comment (IvanGoncharov) [reply README.md]: Wow, you are fast ⚡️ 😄

**Comment by IvanGoncharov** [comment]
I don't see `happyLegacy.js` in this commit you probably forget to do `git add`.
Please squash your commits and add missing files using `git commit --amend` this will allow git to detect that those are renamed files.

**Comment by mkochendorfer** [comment]
Seems to be an issue with the .gitignore file ignoring new files inside the middleware.  Looking into it.

**Comment by IvanGoncharov** [comment]
@mkochendorfer It ignores top-level `middleware` not the `src/middleware`:
https://github.com/APIs-guru/graphql-voyager/blob/24c895560d3e43aed8e7936c069c47e3583402ea/.gitignore#L11

**Comment by mkochendorfer** [comment]
@IvanGoncharov Nope `middleware` is ignoring any files in any path with `middleware` in the path.  That is why the new files I added did not show up when I was commiting.  Fixing the .gitignore in my commit

**Comment by mkochendorfer** [comment]
Had the same issue with `example/middleware/hapiLegacy.js` for example

**Comment by IvanGoncharov** [comment]
@mkochendorfer Ouch, you are right it should be `middleware/`

**Comment by mkochendorfer** [comment]
@IvanGoncharov Should be ready for review again.  Squashed the commits as well

File: .gitignore [hunk 3]
```diff
@@ -8,4 +8,4 @@ stats.json
 npm-debug.log
 
 lib
-middleware
+/middleware
```
Comment (IvanGoncharov) [inline .gitignore]: should be `middleware/`

Comment (mkochendorfer) [reply .gitignore]: I tried that, but it did not work as desired.

Comment (mkochendorfer) [reply .gitignore]: See https://www.atlassian.com/git/tutorials/saving-changes/gitignore#git-ignore-patterns for more info on why

**Comment by IvanGoncharov** [comment]
For some reason, `git` didn't recognize moved files. Can you do `git reset HEAD~1` and commit it one more time.

File: .gitignore (hunk 3 above)
Comment (mkochendorfer) [reply .gitignore]: Even more info: https://stackoverflow.com/questions/3637660/how-to-exclude-file-only-from-root-folder-in-git

**Comment by mkochendorfer** [comment]
@IvanGoncharov Like that is being cased by the fact that the old file is still there still even though the majority of the content was just moved to the hapiLegacy.js files

**Comment by mkochendorfer** [comment]
Re-rolling it after a reset did not make any difference

File: .gitignore (hunk 3 above)
Comment (IvanGoncharov) [reply .gitignore]: Thanks for the patience 🙇
I learned something new today 🎉

File: README.md [hunk 4]
```diff
@@ -161,10 +161,35 @@ app.listen(3001);
...
+    options: {
+      path: '/voyager',
+      endpointUrl: '/graphql'
+    }
+  });
+
+  await server.start();
+};
+
+init();
```
Comment (IvanGoncharov) [inline README.md]: Not a HAPI user but it took me some mental effort to understand that you need `init` just to chain two promisses.
Can it be written like that:
```js
async function startServer() {
  const server = new Hapi.Server({
    port: 3001
  });

  await server.register({
    plugin: voyagerMiddleware,
    options: {
      path: '/voyager',
      endpointUrl: '/graphql'
    }
  });
 
  await server.start();

  return server;
};
 
init();
```

File: example/middleware/hapi.js [hunk 5]
```diff
@@ -3,40 +3,43 @@ const { graphqlHapi } = require('graphql-server-hapi');
...
-  {
-    register: graphqlHapi,
-    options: {
-      path: '/graphql',
-      graphqlOptions: {
-        schema,
-      },
-      route: {
-        cors: true
+const init = async () => {
```
Comment (IvanGoncharov) [inline example/middleware/hapi.js]: Node.js 6.x doesn't support async/await and I worried that generated js files contain them.
Can you please check that TS transpiles async/await?

Comment (mkochendorfer) [reply example/middleware/hapi.js]: This example is based off of https://hapijs.com/tutorials

Comment (mkochendorfer) [reply example/middleware/hapi.js]: And as far as I can tell this is not actually TS at all in the examples.  Just JS

File: README.md (hunk 4 above)
Comment (mkochendorfer) [reply README.md]: There are many different ways to write this code.  I was just using the most basic example from https://hapijs.com/tutorials as the basis

File: example/middleware/hapi.js (hunk 5 above)
Comment (IvanGoncharov) [reply example/middleware/hapi.js]: Looks like hapi supports only node v8+
https://github.com/hapijs/hapi/blob/master/package.json#L18
So not an issue.

Comment (mkochendorfer) [reply example/middleware/hapi.js]: Correct

File: README.md (hunk 4 above)
Comment (IvanGoncharov) [reply README.md]: Makes sense to use something familiar to HAPI community 👍

**Pull request merged**